### PR TITLE
SNET-23503: enable S3 tarball remote URIs

### PR DESCRIPTION
**Pull request by scottcrossen**
This project currently assumes that an Artifactory-like storage backend (see `download-http.go` FKA `download.go`) is used to store the tarball that contains the Ansible playbooks/roles. This pullrequest adds the option to use S3 -- via URI or ARN.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,24 +59,26 @@ a part of `site.yml`'s run. Use the `debug` option to get more insight to the pr
 
 Config file should be in: `/etc/ansible-puller/config.json`, `$HOME/.ansible-puller.json`, `./ansible-puller.json`
 
-| Config Option            | Default                               | Description                                                                             | Required |
```
Comment (jyates) [inline README.md:62]: possible to not reformat everything?

File: download_s3.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	"regexp"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/sirupsen/logrus"
+)
+
+func s3Downloader(context context.Context) (*manager.Downloader, error) {
```
Comment (jyates) [inline download_s3.go:19]: why do all this work and not use something like https://github.com/hashicorp/go-getter ? its actually what we used for HTTP before we needed to support MD5s

File: main.go [hunk 3]
```diff
@@ -81,7 +82,10 @@ func init() {
 	pflag.String("http-proto", "https", "Set to 'http' if necessary")
 	pflag.String("http-user", "", "HTTP username for pulling the remote file")
 	pflag.String("http-pass", "", "HTTP password for pulling the remote file")
+
 	pflag.String("http-url", "", "Remote endpoint to retrieve the file from")
+	pflag.String("s3-uri", "", "Remote endpoint in S3 to retrieve the file from if s3-arn isn't used")
```
Comment (jyates) [inline main.go:87]: why not just arn? its a bit more verbose, but keeps it simpler

**Review by jyates — COMMENTED** [review]
Seems like could be dramatically simplified with go-getter.

**Comment by scottcrossen** [comment]
Let's move this discussion to https://github.com/teslamotors/ansible_puller/pull/13

File: README.md (hunk 1 above)
Comment (scottcrossen) [reply README.md:62]: This actually removes the "Required" column because that column would be now blank

File: main.go (hunk 3 above)
Comment (scottcrossen) [reply main.go:87]: As opposed to s3-arn or as opposed to also including s3-uri ?

File: download_s3.go (hunk 2 above)
Comment (scottcrossen) [reply download_s3.go:19]: Yep! That library would have been helpful to know beforehand. If you prefer I can spend the additional time to use that instead.

**Pull request closed without merging**